src/twisted/protocols/policies.py

```diff
@@ -124,6 +124,9 @@ def connectionLost(self, reason):
         self.factory.unregisterProtocol(self)
         self.wrappedProtocol.connectionLost(reason)
 
+        # Breaking reference cycle between self and wrappedProtocol.
+        self.wrappedProtocol = None
```

Comment:
do we have tests for this change?

Besides updating the existing tests for policies, I was expecting a new test (or update of an existing test) for connectionLost in which of which the docstring describe that on connection lost the wrappedProtocol is None